services/quiz.js

```diff
@@ -1,19 +1,41 @@
 const conjugate = require('./conjugation');
+const utils = require('./spanishUtils');
+var diff = require('fast-diff');
 const verbs = require('../data/verbs.json');
 const pronouns = require('../data/pronouns.json');
 const tenses = require('../data/tenses.json');
-const modeMap = require('../data/mode-map.json');
+const moods = require('../data/moods.json');
 const irregularVerbs = require('../data/irregular-verbs.json');
+var _ = require('underscore');
 
 const FILTER_ALL = 1;
 const FILTER_NONE = 2;
 const FILTER_BYCASE = 3;
 
-
 function chooseRandom(data) {
 	return data[Math.floor(Math.random() * data.length)];
 }
 
+function chooseMood() {
+	return chooseRandom(moods);
+}
+
+function getTenseByName(name) {
+	correct_tense = {};
+	tenses.forEach(function (tense) {
+		if (tense.name===name) {
+			correct_tense = tense;
+		}
+	});
+	return correct_tense
+}
+
+function chooseTense(mood) {
+	tense_name = chooseRandom(mood.tenses);
```

Comment:
@sdee should use consistent naming convention for variables. In javascript it is camelcase for variables. So this should be tenseName